src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -307,6 +299,28 @@ private static bool IsPackageManager(string name, ExecutionContext context)
             return false;
         }
 
+        /// <summary>
+        /// Check if the Path Update feature is enabled for the given session.
+        /// </summary>
+        private static bool IsPathUpdateFeatureEnabled(ExecutionContext context)
+        {
+            // We check only once per session.
+            if (s_pathUpdateFeatureEnabled.TryGetValue(context, out string value))
+            {
+                // The feature is enabled if the value is not null.
+                return value is { };
+            }
+
+            // Disable Path Update if 'EnvironmentProvider' is disabled in the current session, or the current session is restricted.
+            bool enabled = context.EngineSessionState.Providers.ContainsKey(EnvironmentProvider.ProviderName)
```

Comment:
Which brings us back to the question of is the cache worth it? In my opinion, containsKey is much easier than ConditionalWeakTable.